examples/05_vol_geom_wasm_threejs/index.html

```diff
@@ -0,0 +1,330 @@
+		vologram.geometry.setAttribute(
+			"position", 
+			new THREE.Float32BufferAttribute(
+				new Float32Array(vologram.max_blob_size), 3, false
+			)
+		);
+		vologram.geometry.setAttribute(
+			"normal", 
+			new THREE.Float32BufferAttribute(
+				new Float32Array(vologram.max_blob_size), 3, false
+			)
+		);
+		vologram.geometry.setAttribute(
+			"uv",
+			new THREE.Float32BufferAttribute(
+				new Float32Array(vologram.max_blob_size), 2, false
+			)
+		);
+		vologram.geometry.setIndex(
+			new THREE.Uint16BufferAttribute(
+				new Uint16Array(vologram.max_blob_size), 1, false
+			)
+		);
+	}
+
+	function create_vologram_texture() {
+		var texture_video_el = document.getElementById("texture_video_el");
+		vologram.texture = new THREE.VideoTexture( texture_video_el );
+	}
+
+	function create_vologram_mesh() {
+		vologram.material = new THREE.MeshBasicMaterial({
+			color: 0xffffff,
+			map: vologram.texture,
+		});
+
+		vologram.mesh = new THREE.Mesh(vologram.geometry, vologram.material);
+		vologram.mesh.position.set( 0, 0, 0 )
+		// The vologram is mirrored
+		vologram.mesh.scale.set( -1, 1, 1 )
+		// Look at the camera 
+		vologram.mesh.rotation.set( 0, Math.PI, 0 )
+	}
+
+	function create_vologram() {
+		// It is important to create the geometry and texture before the mesh 
+		create_vologram_geometry();
+		create_vologram_texture();
+		create_vologram_mesh();
+		scene.add( vologram.mesh );
+	}
+
+	function create_threejs_camera() {
+		camera = new THREE.PerspectiveCamera(
+        	50,
+        	canvas.clientWidth / canvas.clientHeight,
+        	0.1, 100,
+    	);
+    	camera.position.set(0.0, 1.2, -5.0);
+		camera.lookAt(0.0, 1.2, 0.0);
+    	return camera;
+	}
+
+	function create_threejs_renderer() {
+		renderer = new THREE.WebGLRenderer({
+        	canvas: canvas,
+    	});
+    	renderer.setSize(canvas.clientWidth, canvas.clientHeight);
+		renderer.setClearColor( new THREE.Color(147.0 / 255.0, 149.0 / 255.0, 237.0 / 255.0), 1.0);
+    	return renderer;
+	}
+
+	function create_threejs_scene() {
+		canvas = document.getElementById("canvas");
+		canvas.width = window.innerWidth - 50;
+		canvas.height = window.innerHeight - 50;
+		scene = new THREE.Scene();
+		const scene_camera = create_threejs_camera()
+		scene.add( scene_camera );
+		const scene_renderer = create_threejs_renderer()
+	}
+
+	function init() {
+		create_threejs_scene();
+		{
+			vologram.n_indices = 0;
+			vologram.has_normals = false;
+			vologram.last_frame_loaded = -1;    // To avoid reloading the same frame.
+			vologram.last_keyframe_loaded = -1; // Records if a preceding keyframe was already loaded.
+			vologram.header_ready = false;      // If file is downloaded.
+			vologram.sequence_ready = false;    // If file is downloaded.
+			vologram.ready = false;             // If data is downloaded and vologram is ready to play.
+		}
+
+		console.log("init done");
+		init_done = true;
+
+		return true;
+	}
+
+	function render() {
+		renderer.render(scene, camera);
```

Comment:
one caveat with mirroring the vologram is that we need to switch webgl to render in clockwise winding order. i couldnt see anywhere to do that in three.js on a per-geometry or per-material basis, so last time i just set it globally in webgl. if that's required we probably want to put the vologram in its own scene, and any other geometry in a normal CCW scene and render them one after the other.